Assets/LoggingManager/LogStore.cs

```diff
@@ -0,0 +1,276 @@
+﻿using System;
+using System.CodeDom;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Boo.Lang.Runtime;
+using UnityEngine;
+
+
+public enum LogType
+{
+    Normal,
+    Meta
+}
+
+public class LogStore : MonoBehaviour
+{
+
+    private SortedDictionary<string, List<string>> logs;
+    private StringBuilder logString;
+    public string Label { get; set; }
+
+    private LogType logType;
+    private bool logCommonColumns;
+
+    private int nbLines;
+    private bool createStringOverTime;
+    private StringBuilder currentLineLogged;
+    private SortedDictionary<string, string> currentLogs;
+    private const string fieldSeparator = ";";
+    private const string lineSeperator = "\n";
+    private string email;
+    public string SessionId { get; set; }
+
+    public LogStore(string label, string email, bool createStringOverTime,
+        LogType logType = LogType.Normal, bool logCommonColumns = true)
+    {
+        Init(label, email, Guid.NewGuid().ToString(), createStringOverTime, logType, logCommonColumns);
+    }
+
+    public LogStore(string label, string email, string sessionID, bool createStringOverTime,
+        LogType logType = LogType.Normal, bool logCommonColumns = true)
+    {
+        Init(label, email, sessionID, createStringOverTime, logType, logCommonColumns);
+    }
```

Comment:
I guess you are catering to the use case, that someone might want to have a `LogStore` even if they are not using `LoggingManager` by having two separate constructors? or what is the reason?

Reply:
There are 2 constructors in this class to have the possibility of giving a custom sessionID when creating the LogStore object.
If one is given it will be used, otherwise it will be generated. 

I think I'll edit the `GenerateUIDs` [method](https://github.com/Vincent-SD/ArduinoLogger/blob/06cbd9a1d1288205624b9d286d0c12b86799b5c6/Assets/LoggingManager/LoggingManager.cs#L69) to use the `Guid.NewGuid()` function instead of the current manual way of creating the IDs. 
The ID generated with this method is the one used in each log created in the session.

Reply:
Ah ok. And in which use case do we need to provide a custom sessionID?

Reply:
We need to make sure that every log in the same session has the same ID (meta logs and regular logs for instance) and we have to generate a new one when starting another session. 

In our use case, the ID has to be defined in the LoggingManager and then provided to each LogStore object (using the [constructor](https://github.com/Vincent-SD/ArduinoLogger/blob/06cbd9a1d1288205624b9d286d0c12b86799b5c6/Assets/LoggingManager/LogStore.cs#L41) that has the ID in its parameter).

Actually, the LogStore constructor that generates the sessionID is never used, I just decided to put this here in case one day (again) this class is no longer used with this LoggingManager but in another project.

Reply:
OK - well, if we by design generate IDs outside LogStore, I think we should just keep a single constructor.
Other projects could always pass `Guid.NewGuid()` directly as an argument as they initialize the LogStore - for the time being I dont think it warrants having two constructors that are otherwise identical.

Reply:
No problem, I'll just delete it then.